lib/lotus/utils/kernel.rb

```diff
@@ -418,8 +426,9 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Lotus::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: DEFAULT_PRECISION)
         case arg
+        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
```

Comment:
@jodosha for JRuby only Rational has `#to_d` method with one parameter, that parameter is the precision.